src/game_screens/game_screen.py

```diff
@@ -3,116 +3,172 @@
+        if coin is not None:
+            self.coinlist.append(coin)
 
-        for b in self.coinlist[0:self.arbit_var:self.game_manager.difficulty.value["DENSITY"]]:
-            # (use 14 or 15) this is for the rate at which
-            # objects fall, can change this
-            b.draw()
-            b.fall()
-            self.cart.collect_item(b)
+        # drawing coins and checking coin collisions
+        for coin in self.coinlist:
+            coin.draw()
+            coin.fall()
 
-        self.arbit_var += 1
+            if self.cart.check_collision(coin):
+                self.scoring_function(coin)
 
+            if self.death_zone.check_collision(coin):
+                self.coinlist.remove(coin)
+                del coin
+
+        self.cart.move()
         self.cart.draw()
 
+        self.animation_manager.draw_animations()
+
         # Update time
         seconds = self.gameclock.get_time() / 1000.0
         self.timer += seconds
 
         # returns real value of timer to int value
         int_timer = math.trunc(self.timer)
-        self.texts['Score'].change_text('Score: ' + str(self.cart.points))
+        self.texts['Score'].change_text('Score: ' + str(int(self.game_manager.score)))
         self.texts['Time'].change_text('Time: ' + str(int_timer))
 
-        self.pygame.display.flip()
-
         if self.timer > 30 or self.cart.dead:
-            self.need_reset = True
-            self.game_manager.score = self.cart.points
-            return Game_mode.GAME_OVER
+            self.game_manager.score = int(self.game_manager.score)
+            self.waiting_death_explosion = True
 
         for event in events:
             if event.type == QUIT:
                 return Game_mode.QUIT
 
         return Game_mode.GAME
 
-    def get_random_entity(self, arbit_var, res, size, surface):
-        # randomizing bonus coin/bomb/coin fall frequency, can change this
-        if not arbit_var % 3 or not arbit_var % 4:
-            select = random.randint(1, 2)
-            if select == 1:
-                c = BlueCoin(res, size, surface)
-            else:  # select = 2
-                c = Bomb(res, size, surface)
-        elif not arbit_var % 5 or not arbit_var % 7 or not arbit_var % 11:
-            c = Bomb(res, size, surface)
+    def get_random_entity(self):
+
+        if random.random() < self.params['spawn_chance']:
+
+            coin_select_random = random.random()
+            self.new_random_x()
+
+            if coin_select_random < self.params['silver_chance']:
+                return Coin(self.res, self.size, self.surface, self.random_pos_x, -50)
+            elif coin_select_random < (self.params['silver_chance'] + self.params['gold_chance']):
+                return BlueCoin(self.res, self.size, self.surface, self.random_pos_x, -50)
+            else:
+                return Bomb(self.res, self.size, self.surface, self.random_pos_x, -50)
+
+        else:
+            return None
+
+    def new_random_x(self):
```

Comment:
can name this better instead of `new_random_x()`

Reply:
also, ~overlaping~ overlapping

Reply:
Fixed both